overlay.nix

```
@@ -0,0 +1,8 @@
+self: super: {
+    # includes fix for building ansible from 2020-08-25
+    poetry2nix = import super.fetchFromGitHub {
+        owner = "nix-community";
+        repo = "poetry2nix";
+        rev = "c377518ddf06c391b2ef4e424c84e88f007f632d";
```

```suggestion
        rev = "c377518ddf06c391b2ef4e424c84e88f007f632d";
        sha256 = "<hash here>";  # you can set this to super.lib.fakeSha256 temporarily
```

All side effects in Nix need to be reproducible; hence you need to provide the hash of any sources you download.   if you set it.   You can use `nix-prefetch-url` to calculate the hash or temprarily set `sha256 = super.lib.fakeSha256` (which is just all `00000`) and then nix will tell you there's a hash mismatch in the error message.

---

However; instead of doing this manually in the overlay;

why not do `niv add nix-community/poetry2nix` then niv will handle keeping this github dependency up to date.

---

I think I need to resolve this with you during a call. The above syntax also looks very funky. What's the type of things one can import? What type do I get when I say `import ./sources.nix`? Why is an extra `{}` necessary at the end?

Niv complains for me with a 401 unauthorized when I run the `niv add` command.

---

I figured it out here: https://github.com/wireapp/wire-server-deploy/blob/akshaymankar/sft-metrics-discovery/shell.nix